tests/unit/Snak/ExtendedPropertyValueSnakTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Wikibase\DataModel\Tests\Snak;
+
+use DataValues\StringValue;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Snak\ExtendedPropertyValueSnak;
+
+/**
+ * @covers Wikibase\DataModel\Snak\ExtendedPropertyValueSnak
+ *
+ * @group Wikibase
+ * @group WikibaseDataModel
+ * @group WikibaseSnak
+ *
+ * @licence GNU GPL v2+
+ * @author Adam Shorland
+ */
+class ExtendedPropertyValueSnakTest extends SnakObjectTest {
+
+	public function constructorProvider() {
+		return array(
+			'No extras' => array(
+				true,
+				new PropertyId( 'P1' ),
+				new StringValue( 'a' ),
+				array(),
+			),
+			'2 extras' => array(
+				true,
+				new PropertyId( 'P9001' ),
+				new StringValue( 'bc' ),
+				array( 'foo' => new StringValue( 'foo' ), 'bar' => new StringValue( 'bar' ) ),
+			),
+			'fail - Integer key' => array(
+				'InvalidArgumentException',
+				new PropertyId( 'P9001' ),
+				new StringValue( 'bc' ),
+				array( new StringValue( 'foo' ) ),
+			),
+			'fail - not a value' => array(
+				'InvalidArgumentException',
+				new PropertyId( 'P9001' ),
+				new StringValue( 'bc' ),
+				array( 'foo' => 'bar' ),
+			),
+		);
+	}
+
+	public function getClass() {
+		return 'Wikibase\DataModel\Snak\ExtendedPropertyValueSnak';
```

Comment:
It would be nicer to not follow this not so nice testing pattern for this new code here yes.